mixer/cmd/root.go

```diff
@@ -105,6 +108,18 @@ var RootCmd = &cobra.Command{
 		// If so: inform, stage, and exit.
 		// If not: run command in container and cancel pre-run
 		if !cmdContains(cmd, "format-bump") && !cmdContains(cmd, "upstream-format") && cmdContains(cmd, "build") {
+			// --offline=true AND --native=false, try to see if container exists
+			if !networkCheck && !builder.Native {
+				fmt.Println("Warning: Unable to determine upstream format in --offline mode, build may fail if format boundary is crossed.")
+				format := b.State.Mix.Format
+				_, err := b.GetDockerImageName(format)
```

Comment:
Why are you checking this here? It gets checked in `RunCommandInContainer`, which you call directly below.